### DISCUSSION TITLE
Why a cover added after the manhole creation is not visible in `vw_qgep_wastewater_structure`

### DISCUSSION THREAD

**Discussion by rbovard** [teksi/wastewater#376] [Q&A] [answered]
Imagine I delete a cover related to a manhole, I want to recreate it specifing its `fk_wastewater_structure`.

But in this case, even I have only one cover related to this manhole, it doesen't appear in `vw_qgep_wastewater_structure` (empty `co_obj_id`).

**Cover created at the same time as the manhole**

![image](https://github.com/user-attachments/assets/99c7bc3d-4f1a-49bf-a800-628368f166a7)

![image](https://github.com/user-attachments/assets/d86ce75e-bbe8-4799-89d5-9bd614e239e7)

**Cover created after the manhole**

![image](https://github.com/user-attachments/assets/15c93541-7960-4359-97b5-be6ccd1065c9)

![image](https://github.com/user-attachments/assets/2e8844ac-847a-44dd-92a2-902ff0968ac3)

So how can I specify my unique cover as main cover?

**Comment by sjib** [comment]
You have to manually set the fk_main_cover in class wastewater_structure. There is no UI face yet for that.

**Reply by rbovard** [reply]
Okay I see, thanks.

But can't we imagine to add a mechanism to get the FK if there is only 1 cover?

**Reply by sjib** [reply, marked as answer]
It already exists: https://github.com/teksi/wastewater/blob/93f00d1accda9766a25e9fd4d7bc5ab865b9cff2/datamodel/app/symbology_functions.sql#L290

You can run that tww_app.wastewater_structure_update_fk_main_cover manually

_all is a bool variable. The function can be called `tww_app.wastewater_structure_update_fk_main_cover (NULL, True)` to update all or `tww_app.wastewater_structure_update_fk_main_cover (obj_id) `to update a specific entry.

See also https://github.com/teksi/wastewater/issues/264

**Reply by rbovard** [reply]
Thanks! That was exactly what I was looking for, I've added that in my nightly cron.
